src/main/java/frc/team2412/robot/commands/intake/IntakeBitmapCommand.java

```diff
@@ -0,0 +1,98 @@
+                                                                                    // shooter (may change later)
+        H(true, true, true, false, 0, INDEX_IN_SPEED, INDEX_IN_SPEED, true), // correct ball in ingest, wrong ball in
+                                                                                // feeder
+        I(true, true, false, false, 0, INDEX_IN_SPEED, INDEX_IN_SPEED, true); // wrong ball in both
+
+        boolean ingestSensor, feederSensor, ingestColor, feederColor, shooterMisfire;
+        double intakeMotorSpeed, ingestMotorSpeed, feederMotorSpeed;
+
+        private Bitmap(boolean ingestSensor, boolean feederSensor, boolean ingestColor, boolean feederColor,
+                double intakeMotorSpeed, double ingestMotorSpeed, double feederMotorSpeed, boolean shooterMisfire) {
+            this.ingestSensor = ingestSensor;
+            this.feederSensor = feederSensor;
+            this.ingestColor = ingestColor;
+            this.feederColor = feederColor;
+            this.intakeMotorSpeed = intakeMotorSpeed;
+            this.ingestMotorSpeed = ingestMotorSpeed;
+            this.feederMotorSpeed = feederMotorSpeed;
+            this.shooterMisfire = shooterMisfire;
+        }
+
+        public boolean equals(boolean ingestSensor, boolean feederSensor, boolean ingestColor, boolean feederColor) {
+            return this.ingestSensor == ingestSensor &&
+                    this.feederSensor == feederSensor &&
+                    this.ingestColor == ingestColor &&
+                    this.feederColor == feederColor;
+        }
+
+    }
+
+    private IntakeSubsystem intakeSubsystem;
+    private IndexSubsystem indexSubsystem;
+    private ShooterSubsystem shooterSubsystem;
+    private ShooterVisionSubsystem shooterVisionSubsystem;
+
+    public IntakeBitmapCommand(IntakeSubsystem intakeSubsystem, IndexSubsystem indexSubsystem,
+            ShooterSubsystem shooterSubsystem, ShooterVisionSubsystem shooterVisionSubsystem) {
+        this.intakeSubsystem = intakeSubsystem;
+        this.indexSubsystem = indexSubsystem;
+        this.shooterSubsystem = shooterSubsystem;
+        this.shooterVisionSubsystem = shooterVisionSubsystem;
+    }
+
+    @Override
+    public void execute() {
+        boolean ingestSensor = indexSubsystem.ingestSensorHasBallIn();
+        boolean feederSensor = indexSubsystem.feederSensorHasBallIn();
+        boolean ingestColor = indexSubsystem.ingestHasCorrectCargo();
+        boolean feederColor = indexSubsystem.feederHasCorrectCargo();
+
+        for (Bitmap value : Bitmap.values()) {
+
+            if (value.equals(ingestSensor, feederSensor, ingestColor, feederColor)) {
+                intakeSubsystem.setSpeed(value.intakeMotorSpeed);
+                indexSubsystem.setSpeed(value.ingestMotorSpeed, value.feederMotorSpeed);
+
+                if (value.shooterMisfire) {
+                    shooterSubsystem.setHoodAngle(0);
```

Comment:
Even with a very small velocity, a hood angle of 0 still raises the possibility of accidentally shooting off the field if we get rammed close to the side wall.

Reply:
ideally the shooter should always be facing the hub so it shouldnt be a problem? we will test this later

Reply:
While technically facing the hub, an angle of 0 means the ball will shoot straight up in the air so it isn't really facing the hub.  Seems like a small angle (5-10) may help prevent balls accidentally being shot out of the field, but still not be in danger of being launched into the lower goal.  Testing could help determine, but I am not sure how excited we are to try ramming another robot into ours during the testing phase.